lib/server.js

```diff
@@ -0,0 +1,70 @@
+
+/**
+ * Module dependencies.
+ */
+
+var _ = require('lodash');
+var AuthenticateHandler = require('./handlers/authenticate-handler');
+var AuthorizeHandler = require('./handlers/authorize-handler');
+var InvalidArgumentError = require('./errors/invalid-argument-error');
+var TokenHandler = require('./handlers/token-handler');
+
+/**
+ * Constructor.
+ */
+
+function OAuth2Server(options) {
+  options = options || {};
+
+  if (!options.model) {
+    throw new InvalidArgumentError('Missing parameter: `model`');
+  }
+
+  this.options = _.assign({
+    accessTokenLifetime: 60 * 60,
+    authCodeLifetime: 5 * 60,
+    refreshTokenLifetime: 1209600
```

Comment:
User multiplication as well. A comment specifying the time unit would also be welcome :-)